tdrs-backend/tdpservice/scheduling/parser_task.py

```diff
@@ -0,0 +1,19 @@
+"""Celery hook for parsing tasks."""
+from __future__ import absolute_import
+from celery import shared_task
+import logging
+from tdpservice.data_files.models import DataFile
+from tdpservice.search_indexes.parsers.preparser import preparse
+
+logger = logging.getLogger(__name__)
+
+@shared_task
+def parse(data_file_id):
```

Comment:
I am a little confused by the retry functionality. What changes between tries? Is this meant to account for an asynchronous function not sending the entire payload?

Reply:
I think this could be a good idea if done in the right way after #1354. Really, I'd like to see if perhaps we can trigger a retry via the django admin console. The reason I say after that ticket is because right now we throw exceptions when long term we will just be storing everything to the database. So that ticket will provide better clarification between a user-facing error and a systems-error that I think the retry functionality should handle. This is a great idea though for probably most of our celery stuff.


Unrelated, I wouldn't be opposed to shoe-horning in this exception 😆 
`from twitter.exceptions import FailWhaleError`